python/mxnet/optimizer.py

```diff
@@ -80,6 +71,20 @@ def set_lr_scale(self, args_lrscale):
         """
         self.lr_scale = args_lrscale.copy()
 
+    def _update_count(self, index):
+        """
+        update num_update
+
+        Parameters:
+        index : int
+            The index will be updated
+        """
+        if index not in self._index_update_count:
+            self._index_update_count[index] = 0
+        self._index_update_count[index] += 1
+        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
I am not sure about this `max`. In ASGD settings, it is just really weird to let each worker decrease learning rate separately. So it is possible that, some workers are using lr = 0.5 while some workers are using lr = 0.005. What does this mean in fact?

I think a (slightly) better semantic is to tag the weight with version and the learning rate strategy is version-based. Say if the version is larger than some step, decrease the learning rate to some extent.